### DISCUSSION TITLE
OKE support (for workload clusters)

### DISCUSSION THREAD

**Discussion by MPV** [oracle/cluster-api-provider-oci#101] [Ideas]
Does CAPOCI already come with OKE support (for workload clusters), or are there any plans for it?

**Comment by Jenksho** [comment]
Hi Victor,
we have it in plan but would love to know more about your use-case if you could share and we can prioritise as necessary.

**Reply by MPV** [reply]
The idea would be to rent the control-plane instead of the cognitive load/overhead/cost of managing it oneself.

Sticking with ClusterAPI (over Terraform/Crossplane etc) would enable consistency across clusters and their management for those using CAPI for different providers.

For comparison, check out CABPE (Cluster API bootstrap provider EKS):

- https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/bootstrap/eks

**Comment by joekr** [comment]
Investigation issue https://github.com/oracle/cluster-api-provider-oci/issues/110

**Comment by shyamradhakrishnan** [comment]
@MPV we have a proposal here - https://docs.google.com/document/d/10aDgpZk7OLswVV5e0cIK9GXuckEOQfsT1-MkQDk5X_k and is working on an implementation
Please review if possible, please note OKE cannot import any node into a cluster, so both control plane and machines(via nodepool) will have to be used.
